distros/spring/src/main/java/com/connexta/transformation/service/PollResponseCreator.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.service;
+
+import com.connexta.transformation.commons.api.ErrorCode;
+import com.connexta.transformation.commons.api.status.MetadataTransformation;
+import com.connexta.transformation.commons.api.status.Transformation;
+import com.connexta.transformation.commons.api.status.TransformationStatus.State;
+import com.connexta.transformation.rest.models.ErrorMessage;
+import com.connexta.transformation.rest.models.MetadataInformation;
+import com.connexta.transformation.rest.models.MetadataType;
+import com.connexta.transformation.rest.models.Status;
+import com.connexta.transformation.rest.models.TransformationPollResponse;
+import com.connexta.transformation.service.Exceptions.UnknownStateException;
+import java.net.URI;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * PollResponseCreator is a method object that is used to create a {@link
+ * TransformationPollResponse} with the information provided by a {@link Transformation}.
+ */
+class PollResponseCreator {
+
+  private static final Map<State, Status> statusMapping =
```

Comment:
I'm wary of doing this because it would couple the `TransformationManager` to the OpenAPI API.

Reply:
Fair enough. Although keeping a mapping table like that makes it very easy to forget to update it which will translate in `null`. Let's make sure we account for missing entries by using `getOrDefault()` and map to an unknown error back instead of throwing back an `IllegalStateException`

❗️ BTW, the attribute should be `STATUS_MAPPINGS` since it is a constant.